MH4U Dex/ItemEncyclopediaTableViewController.m

```diff
@@ -61,11 +57,39 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
     return cell;
 }
 
+- (NSArray *)sectionIndexTitlesForTableView:(UITableView *)tableView
+{
+    //return [[UILocalizedIndexedCollation currentCollation] sectionIndexTitles];
+    return [self.fetchedResultsController sectionIndexTitles];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView sectionForSectionIndexTitle:(NSString *)title atIndex:(NSInteger)index
+{
+    return [self.fetchedResultsController sectionForSectionIndexTitle:title atIndex:index];
+}
+
 #pragma mark - Helper Methods
 
 - (Item *)itemAtIndexPath:(NSIndexPath *)indexPath
 {
-    return self.items[indexPath.row];
+    Item *item = [self.fetchedResultsController objectAtIndexPath:indexPath];
+    return item;
+}
+
+- (NSFetchedResultsController *)fetchedResultsControllerWithPredicate:(NSPredicate *)predicate
+{
+    NSError *fetchError = nil;
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] initWithEntityName:[Item entityName]];
+    fetchRequest.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:ItemAttributes.name ascending:YES]];
+    fetchRequest.fetchBatchSize = 20;//TODO: Tinker with this.
+    fetchRequest.predicate = predicate;
+    NSFetchedResultsController *newResultsController = [[NSFetchedResultsController alloc] initWithFetchRequest:fetchRequest managedObjectContext:[CoreDataController sharedCDController].managedObjectContext sectionNameKeyPath:@"firstLetter" cacheName:@"Root"];
```

Comment:
It's usually a good idea to avoid hard-coding method or property names like "firstLetter" here. If you make the `firstLetter` method public, you could then use this for the `sectionNameKeyPath` param:

```
NSStringFromSelector(@selector(firstLetter))
```